pkg/arvo/sys/vane/jael.hoon

```diff
@@ -733,15 +740,18 @@
     =/  a-point=point  (~(gut by pos.zim.pki) ship.i.udiffs *point)
     =/  a-diff=(unit diff:point)  (udiff-to-diff:point udiff.i.udiffs a-point)
     =?  this-su  ?=(^ a-diff)
-      ::  if this about our keys, and we already know these, start using them
-      ::
-      =?  lyf.own
-          ?&  =(our ship.i.udiffs)
+      =-  (public-keys:feel original-pos %diff ship.i.udiffs u.a-diff)
+      ?.  ?&  =(our ship.i.udiffs)
               ?=(%keys -.u.a-diff)
               (~(has by jaw.own) life.to.u.a-diff)
           ==
-        life.to.u.a-diff
-      (public-keys:feel original-pos %diff ship.i.udiffs u.a-diff)
+        this-su
+      ::  if this about our keys, and we already know these, start using them
+      ::
+      =.  lyf.own  life.to.u.a-diff
+      ::  notify subscribers (ames) to start using our new private keys
+      ::
+      (exec yen.own [%give %private-keys [lyf jaw]:own])
```

Comment:
Does this work?  I think so, but the pattern feels weird since it puts two conflicting copies of `this-su` in our subject, like as though it needs a `public-keys:feel:-` to make sure we're using the right one.

I'd probably do something like:

```
      =?    this-su
          ?&  =(our ship.i.udiffs)
              ?=(%keys -.u.a-diff)
              (~(has by jaw.own) life.to.u.a-diff)
          ==
        ::  if this about our keys, and we already know these, start using them
        ::
        =.  lyf.own  life.to.u.a-diff
        ::  notify subscribers (ames) to start using our new private keys
        ::
        (exec yen.own [%give %private-keys [lyf jaw]:own])
      ::
      (public-keys:feel original-pos %diff ship.i.udiffs u.a-diff)
```

Reply:
Yeah, I tested it with a couple of L2 ships, but agree that your suggestion looks better—I deployed it https://github.com/urbit/urbit/pull/5532/commits/432d9674f5cb780f1de2a33f88e3b07404b632b4 on `~sordem` and also booted a L2 ship from a the pill and checking that communication with other ships works fine before and after switching to the new private keys.